sonar-scanner-engine/src/main/java/org/sonar/scanner/issue/tracking/ServerIssueRepository.java

```diff
@@ -45,21 +45,20 @@
   private final Storages caches;
   private Storage<ServerIssue> issuesCache;
   private final ServerIssuesLoader previousIssuesLoader;
-  private final ImmutableProjectReactor reactor;
-  private final InputComponentStore resourceCache;
+  private final InputComponentStore componentStore;
 
-  public ServerIssueRepository(Storages caches, ServerIssuesLoader previousIssuesLoader, ImmutableProjectReactor reactor, InputComponentStore resourceCache) {
+  public ServerIssueRepository(Storages caches, ServerIssuesLoader previousIssuesLoader, InputComponentStore componentStore) {
     this.caches = caches;
     this.previousIssuesLoader = previousIssuesLoader;
-    this.reactor = reactor;
-    this.resourceCache = resourceCache;
+    this.componentStore = componentStore;
   }
 
   public void load() {
     Profiler profiler = Profiler.create(LOG).startInfo(LOG_MSG);
     this.issuesCache = caches.createCache("previousIssues");
     caches.registerValueCoder(ServerIssue.class, new ServerIssueValueCoder());
-    previousIssuesLoader.load(reactor.getRoot().getKeyWithBranch(), this::store);
+    DefaultInputModule root = (DefaultInputModule) componentStore.root();
+    previousIssuesLoader.load(root.getKeyWithBranch(), this::store);
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') A "NullPointerException" could be thrown; "root" is nullable here. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS2259)